conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/DoubleArrayList.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib.internal;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.RandomAccess;
+
+public final class DoubleArrayList extends AbstractList<Double> implements RandomAccess {
+    private static final double[] EMPTY_ELEMENTDATA = {};
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+    private int size = 0;
+    private double[] elements;
+
+    public DoubleArrayList() {
+        this(16);
+    }
+
+    private void resizeIfNecessary(int toAdd) {
+        int minCapacity = size + toAdd;
+        if (minCapacity < 0) {
+            throw new OutOfMemoryError();
```

Comment:
These OutOfMemory errors were pulled from the ArrayList implementation. Kept it like this for consistency but also not opposed to using a safe runtime exception or something.